datamol/utils/file_load_save.py

```diff
@@ -0,0 +1,99 @@
+from typing import cast
+
+import pandas as pd
+import datamol as dm
+
+EXTENSIONS_DICT = {
+    "csv": [
+        ".csv",
+        ".csv.gz",
+        ".csv.bz2",
+        ".csv.zip",
+        ".csv.xz",
+        ".csv.zst",
+        ".csv.tar",
+        ".csv.tar.gz",
+        ".csv.tar.xz",
+        ".csv.tar.bz2",
+    ],
+    "excel": [".xlsx"],
+    "parquet": [".parquet"],
+    "json": [
+        ".json",
+        ".json.gz",
+        ".json.bz2",
+        ".json.zip",
+        ".json.xz",
+        ".json.zst",
+        ".json.tar",
+        ".json.tar.gz",
+        ".json.tar.xz",
+        ".json.tar.bz2",
+    ],
+    "sdf": [".sdf"],
```

Comment:
Can you check whether `.sdf.gz` files are supported and add it if that's the case?

See https://www.rdkit.org/docs/GettingStartedInPython.html#reading-sets-of-molecules